src/contentmanager.cpp

```diff
@@ -406,6 +406,8 @@ QStringList ContentManager::getBookIds()
 
     filter.acceptTags(tags);
     filter.query(m_searchQuery.toStdString());
+    if(m_currentLanguage != QLatin1Char('*'))
+        filter.lang(m_currentLanguage.toStdString());
```

Comment:
I wonder why `QLatin1Char()` is necessary? please put a space after the `if`.

Reply:
We're compare QString to QChar, QLatin1Char ensure the '*' is constructed at compile time, no runtime cost. Otherwise QChar will be contructed from '*' (char) at runtime.

Reply:
I don't see any clue that `QLatin1Char` is build as compile time. (https://doc.qt.io/qt-5/qlatin1char.html)
What is probably done, is that we are creating a QLatin1Char, converting it to a QChar (using the implicit constructor), converting it to a QString(using the implicit constructor) then comparing the QString using `operator!=`.

We should probably use `QStringLiteral` to construct a QString at compile time or even better `QLatin1String` as `operator!=` is overloaded for `QLatin1String` (https://doc.qt.io/qt-5/qstring.html#QStringLiteral)

Reply:
Sorry for being misleading, you're right. Since the QString uses char * internally, and provides const char * overload for !=, I switched to just plain c char array.

Reply:
To be totally thorough, the `!=` operator convert the char* to a QString internally.
But we are not in a critical path, the conversion will have no impact. It is simpler to keep the plain `char*`.

Reply:
Qt doc says QString accepts const char * to avoid construct QString [there](https://doc.qt.io/qt-5/qlatin1string.html#details)
But it [constructs QString nonetheless](https://doc.qt.io/qt-5/qstring.html#operator-not-eq-1)

:man_facepalming: